lib/config/core.js

```diff
@@ -353,9 +353,15 @@ Conf.prototype.addEnv = function (env) {
 
       // leave first char untouched, even if
       // it is a '_' - convert all other to '-'
+      // except for if there is :_ or :__ as
+      // these are used for _password and __auth
       var p = k.toLowerCase()
-               .replace(/^npm_config_/, '')
-               .replace(/(?!^)_/g, '-')
+              .replace(/^npm_config_/, '')
+              .replace(/:_/, 'PLACEHOLDERONE')
+              .replace(/:__/, 'PLACEHOLDERTWO')
```

Comment:
this second regexp will never match because `:__ABC` will be replaced with `PLACEHOLDERONE_ABC`